### PR TITLE
feat: migrate to meros

### PR DESCRIPTION
**Pull request by maraisr**
This migrates the project in using `meros`.

This has uncovered some not-so-nice DX in using it 😔 But I will PR once I correct this.

- Mainly around TypeScript support and handling that union between Response and AsyncGenerator. 
- Import from `import { meros } from 'meros';` works, but TypeScript coundnt find types.
- Also thinking a callback add-on might be handy for places that can't be async.

But will resolve all of that over at meros, for the meantime this works, and works great!

### CODE REVIEW

File: src/dev/GraphiQL.tsx [hunk 1]
```diff
@@ -67,25 +67,42 @@ const httpFetcher: Fetcher = (graphQLParams) =>
...
+              "accept": "application/json, multipart/mixed",
+              "content-type": "application/json",
+            },
+            signal: abortController.signal,
+          });
+
+          const patches = await meros(response);
+          // @ts-ignore we need to ignore coz typescript don't understand this.
+          if (patches[Symbol.asyncIterator] < 'u') {
+            for await (const patch of patches) {
```
Comment (sibelius) [inline src/dev/GraphiQL.tsx:89]: nice api

File: src/dev/GraphiQL.tsx [hunk 2]
```diff
@@ -67,25 +67,42 @@ const httpFetcher: Fetcher = (graphQLParams) =>
 const httpMultipartFetcher: Fetcher = (graphQLParams) =>
   ({
     subscribe: (...args: SubscribeArguments) => {
+      const abortController = new AbortController();
       const sink = getSinkFromArgs(args);
       const isIntrospectionQuery = args.length === 3;
```
Comment (maraisr) [inline src/dev/GraphiQL.tsx:72]: whoops, think we can get rid of this
```suggestion
```

File: src/dev/GraphiQL.tsx [hunk 3]
```diff
@@ -67,25 +67,42 @@ const httpFetcher: Fetcher = (graphQLParams) =>
...
-        onNext: (parts) => {
-          // Introspection is broken if we return a array instead of a single item.
-          // TODO: This should be addressed inside GraphiQL
-          sink.next(isIntrospectionQuery ? parts[0] : parts);
-        },
-        onError: sink.error,
-        onComplete: sink.complete,
-      });
+      try {
+        (async () => {
```
Comment (n1ru4l) [inline src/dev/GraphiQL.tsx:75]: Will this be caught without await-ing the promise returned by the IIFE?

File: src/dev/GraphiQL.tsx (hunk 2 above)
Comment (n1ru4l) [reply src/dev/GraphiQL.tsx:72]: Eh, this is for an inconsistency issue inside graphiql. See https://github.com/graphql/graphiql/pull/1639 and https://github.com/graphql/graphiql/issues/1638

Comment (n1ru4l) [reply src/dev/GraphiQL.tsx:72]: ah nervermind :D

Comment (n1ru4l) [reply src/dev/GraphiQL.tsx]: I applyied the change via commit suggestion :)

File: src/dev/GraphiQL.tsx (hunk 1 above)
Comment (n1ru4l) [reply src/dev/GraphiQL.tsx:88]: Can we please replace `if (patches[Symbol.asyncIterator] < 'u') {` with a check that is clear without reading up on it 😅. For demonstration purposes, I wanna keep it as easy to understand as possible :)

**Pull request merged**